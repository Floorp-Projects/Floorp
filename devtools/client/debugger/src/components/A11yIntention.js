/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import "./A11yIntention.css";

export default class A11yIntention extends React.Component {
  state = { keyboard: false };

  handleKeyDown = () => {
    this.setState({ keyboard: true });
  };

  handleMouseDown = () => {
    this.setState({ keyboard: false });
  };

  render() {
    return (
      <div
        className={this.state.keyboard ? "A11y-keyboard" : "A11y-mouse"}
        onKeyDown={this.handleKeyDown}
        onMouseDown={this.handleMouseDown}
      >
        {this.props.children}
      </div>
    );
  }
}
