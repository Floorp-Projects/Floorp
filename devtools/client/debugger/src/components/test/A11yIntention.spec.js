/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";
import A11yIntention from "../A11yIntention";

function render() {
  return shallow(
    <A11yIntention>
      <span>hello world</span>
    </A11yIntention>
  );
}

describe("A11yIntention", () => {
  it("renders its children", () => {
    const component = render();
    expect(component).toMatchSnapshot();
  });

  it("indicates that the mouse or keyboard is being used", () => {
    const component = render();
    expect(component.prop("className")).toEqual("A11y-mouse");

    component.simulate("keyDown");
    expect(component.prop("className")).toEqual("A11y-keyboard");

    component.simulate("mouseDown");
    expect(component.prop("className")).toEqual("A11y-mouse");
  });
});
