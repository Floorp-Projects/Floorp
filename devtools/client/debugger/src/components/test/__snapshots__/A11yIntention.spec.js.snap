// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A11yIntention renders its children 1`] = `
<div
  className="A11y-mouse"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
>
  <span>
    hello world
  </span>
</div>
`;
